modules/unspents/src/dimensions.ts

```diff
@@ -290,434 +154,392 @@ Object.defineProperty(Dimensions.prototype, 'nInputs', {
+  }
 
-/**
- * @param chain
- * @return {Number}
- */
-Dimensions.getOutputScriptLengthForChain = function (chain: ChainCode): number {
-  switch (scriptTypeForChain(chain)) {
-    case 'p2wsh':
-    case 'p2tr':
-      return 34;
-    default:
-      return 23;
+  /**
+   * @param chain
+   * @return {Number}
+   */
+  static getOutputScriptLengthForChain(chain: ChainCode): number {
+    switch (scriptTypeForChain(chain)) {
+      case 'p2wsh':
+      case 'p2tr':
+        return 34;
+      default:
+        return 23;
+    }
   }
-};
 
-/**
- * @param scriptLength
- * @return {Number} vSize of an output with script length
- */
-Dimensions.getVSizeForOutputWithScriptLength = function (scriptLength: number): number {
-  if (!PositiveInteger.is(scriptLength)) {
-    throw new TypeError(`expected positive integer for scriptLength, got ${scriptLength}`);
+  /**
+   * @param scriptLength
+   * @return {Number} vSize of an output with script length
+   */
+  static getVSizeForOutputWithScriptLength(scriptLength: number): number {
+    if (!PositiveInteger.is(scriptLength)) {
+      throw new TypeError(`expected positive integer for scriptLength, got ${scriptLength}`);
+    }
+    return scriptLength + compactSize(scriptLength) + VirtualSizes.txOutputAmountSize;
   }
-  return scriptLength + compactSize(scriptLength) + VirtualSizes.txOutputAmountSize;
-};
 
-/**
- * @param input - the transaction input to count
- * @param params
- *        [param.assumeUnsigned] - default type for unsigned input
- */
-Dimensions.fromInput = function (input: utxolib.TxInput, params = {}) {
-  const p2shInput = Dimensions.sum({ nP2shInputs: 1 });
-  const p2shP2wshInput = Dimensions.sum({ nP2shP2wshInputs: 1 });
-  const p2wshInput = Dimensions.sum({ nP2wshInputs: 1 });
-  const p2trKeypathInput = Dimensions.sum({ nP2trKeypathInputs: 1 });
-  const p2trScriptPathLevel1Input = Dimensions.sum({ nP2trScriptPathLevel1Inputs: 1 });
-  const p2trScriptPathLevel2Input = Dimensions.sum({ nP2trScriptPathLevel2Inputs: 1 });
-  const p2shP2pkInput = Dimensions.sum({ nP2shP2pkInputs: 1 });
-
-  if (input.script?.length || input.witness?.length) {
-    const parsed = utxolib.bitgo.parseSignatureScript(input);
-    switch (parsed.scriptType) {
+  /**
+   * @param input - the transaction input to count
+   * @param params
+   *        [param.assumeUnsigned] - default type for unsigned input
+   */
+  static fromInput(input: utxolib.TxInput, params: FromInputParams = {}): Dimensions {
+    const p2shInput = Dimensions.sum({ nP2shInputs: 1 });
+    const p2shP2wshInput = Dimensions.sum({ nP2shP2wshInputs: 1 });
+    const p2wshInput = Dimensions.sum({ nP2wshInputs: 1 });
+    const p2trKeypathInput = Dimensions.sum({ nP2trKeypathInputs: 1 });
+    const p2trScriptPathLevel1Input = Dimensions.sum({ nP2trScriptPathLevel1Inputs: 1 });
+    const p2trScriptPathLevel2Input = Dimensions.sum({ nP2trScriptPathLevel2Inputs: 1 });
+    const p2shP2pkInput = Dimensions.sum({ nP2shP2pkInputs: 1 });
```

Comment:
Should these be statics or constants defined outside this function? It seems a bit unnecessary to redeclare these (and go through the steps of constructing new objects in `sum`) each time `fromInput` is called.

Reply:
yes let's clean it up while we are here